File: api/errors/defaults/base_error.py
from fastapi import HTTPException

from api.enums.system import ErrorCodeEnum


class BaseError(Exception):
    def __init__(self, code: ErrorCodeEnum, message: str) -> None:
        super().__init__(message)

        self.code = code
        self.message = message

    def to_http_exception(self, status_code: int) -> HTTPException:
        return HTTPException(
            detail={
                "code": self.code.value,
                "message": self.message,
            },
            status_code=status_code,
        )

File: web/errors/_base/BaseError.ts
// enums
import { ErrorCodeEnum } from '@/enums/common';

export default abstract class BaseError extends Error {
  // private
  private readonly __isQernelError = true;
  // public
  public readonly code: ErrorCodeEnum;
  public readonly displayName: string;
  public message: string;

  public constructor(message: string) {
    super(message.toLowerCase());
  }

  public isQernelError(): boolean {
    return this.__isQernelError;
  }
}
